### DISCUSSION TITLE
How to access raw content from a private org repo (behind SSO) using OAuth access token

### DISCUSSION THREAD

**Discussion by cccs-eric** [community/community#24744] [Apps, API and Webhooks]
<p>Hi,</p>
<p>I am trying to fetch the content of a file using a ‘raw’ URL (<a href="https://raw.githubusercontent.com" rel="noopener nofollow ugc">https://raw.githubusercontent.com</a>).  That file is from a private repository in an Organization using SSO.  I can fetch the file using a Personal Access Token (PAT), but not using an OAuth access token.</p>
<p>Here is what I did:</p>
<ol>
<li>I created a PAT and enabled SSO for that token.(<code>GITHUB_PAT_TOKEN=ghp_MNxHStN2MYXzD..........GI3zSBZ9</code>)</li>
<li>I created an OAuth application in my organization (to be transparent, I first created it under my account and transferred ownership to my org afterward)</li>
<li>I requested an OAuth token using my registered application clientID and secret. (<code>GITHUB_OAUTH_TOKEN=gho_bosr611HZ1Don..........VO0XWMr7</code>)</li>
</ol>
<p>I can hit <code>https://api.github.com/user</code> using both PAT or OAuth token and I get a valid JSON describing my user:</p>
<pre><code class="lang-bash">curl  -X GET -H "Authorization: token $GITHUB_PAT_TOKEN"   -H "Accept: application/vnd.github.v3+json"   https://api.github.com/user
curl  -X GET -H "Authorization: token $GITHUB_OAUTH_TOKEN"   -H "Accept: application/vnd.github.v3+json"   https://api.github.com/user
</code></pre>
<p>So this tells me that my PAT and OAuth tokens are valid.</p>
<p>When I try to fetch a file from a private repository, only the call using the PAT works:</p>
<pre><code class="lang-bash"># No Authorization header
curl  -X GET https://raw.githubusercontent.com/MyOrg/MyPrivateRepo/HEAD/file.yaml
404: Not Found

# Using Authorization header with OAuth access token
curl  -X GET -H "Authorization: token $GITHUB_OAUTH_TOKEN" https://raw.githubusercontent.com/MyOrg/MyPrivateRepo/HEAD/file.yaml
404: Not Found

# Using Authorization header with PAT
curl  -X GET -H "Authorization: token $GITHUB_PAT_TOKEN" https://raw.githubusercontent.com/MyOrg/MyPrivateRepo/HEAD/file.yaml
&lt;content of file.yaml&gt;
</code></pre>
<p>What’s missing to make this work using my OAuth access token?  Is it something even possible?</p>
<p>Note: To give more context on what I am trying to do, I have an application using Keycloak as an Identity Broker and GitHub is a configured Identity Provider.  The OAuth application that I registered at the start of this post is for Keycloak.  A user can authenticate to my application using GitHub as the identity provider.  Once authenticated, Keycloak stores the token returned by GitHub and the user can retrieve it on demand to access files in GitHub.</p>

**Comment by cccs-eric** [comment, marked as answer]
<p>The solution to my problem was a missing scope when requesting an OAuth token.  I needed to add <code>repo</code> as scope.  By default, it was only <code>user:email</code>.</p>

**Comment by aryaddamra-maker** [comment]
<script src="https://gist.github.com/hnnazm/ac6f986d45556e52334fb7fd2689d9be.js"></script>

**Comment by aryaddamra-maker** [comment]
<script src="https://gist.github.com/hnnazm/ac6f986d45556e52334fb7fd2689d9be.js"></script>
